### PR TITLE
Enable AI Chat history feature

### PR DESCRIPTION
**Pull request by petemill**
- Resolves https://github.com/brave/brave-browser/issues/42854

Adds requests from security and privacy review in order to accept the feature being enabled by default:
- Resolves https://github.com/brave/brave-browser/issues/42853

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by fallaciousreasoning — APPROVED** [review]
:tada:

**Comment by diracdeltas** [comment]
This should probably be done before this is merged? https://github.com/brave/reviews/issues/1703#issuecomment-2539690183 (please check with Pat if unclear)

**Comment by petemill** [comment]
The link in the notice now goes directly to the newly published sub-page about Storage https://support.brave.com/hc/en-us/articles/32663367857549-How-do-I-use-Chat-History-in-Brave-Leo

File: components/ai_chat/resources/page/components/notices/notice_conversation_storage.tsx [hunk 1]
```diff
@@ -51,7 +28,7 @@ export default function NoticeConversationStorage() {
           <a
             href='#'
             target='_blank'
-            onClick={() => aiChatContext.uiHandler?.openModelSupportUrl()}
+            onClick={() => aiChatContext.uiHandler?.openStorageSupportUrl()}
```
Comment (diracdeltas) [inline components/ai_chat/resources/page/components/notices/notice_conversation_storage.tsx:31]: ++

Comment (diracdeltas) [reply components/ai_chat/resources/page/components/notices/notice_conversation_storage.tsx:31]: minor suggestion "Chat history is available in Brave version 1.75x and above" should add "on Desktop"

**Comment by diracdeltas** [comment]
maybe not specific to this PR, but this seems buggy:

1. enable AI history feature in nightly (1.75.95 for me)
2. open leo, disable page context
3. it shows me some suggestions but the icon next to them is blank?
Screenshot 2024-12-12 at 13 55 45
4. click on one of them, like 'formulate a business proposal'. notice the output.
5. close leo and reopen
6. click 'conversation history' > 'formulate a business proposal'
7. notice the conversation is blank

File: components/ai_chat/resources/page/components/notices/notice_conversation_storage.tsx (hunk 1 above)
Comment (petemill) [reply components/ai_chat/resources/page/components/notices/notice_conversation_storage.tsx:31]: ...and android...

Comment (diracdeltas) [reply components/ai_chat/resources/page/components/notices/notice_conversation_storage.tsx:31]: so we should specify desktop and android unless we also think it's going to land in iOS 1.75x

**Comment by petemill** [comment]
> maybe not specific to this PR, but this seems buggy:
> 
> 1. enable AI history feature in nightly (1.75.95 for me)
> 2. open leo, disable page context
> 3. it shows me some suggestions but the icon next to them is blank? UPDATE: seems this is an apple problem
> 
> Screenshot 2024-12-12 at 13 55 45
> 4. click on one of them, like 'formulate a business proposal'. notice the output.
> 5. close leo and reopen
> 6. click 'conversation history' > 'formulate a business proposal'
> 7. notice the conversation is blank

@diracdeltas mentioned that emojis aren't working on any website for her, so I think we can ignore that part.

As for the second issue - the cause is:
- https://github.com/brave/brave-browser/issues/42871

File: components/ai_chat/resources/page/components/notices/notice_conversation_storage.tsx (hunk 1 above)
Comment (petemill) [reply components/ai_chat/resources/page/components/notices/notice_conversation_storage.tsx:31]: That edit will be made

**Comment by diracdeltas** [comment]
no security comments from me, will let the privacy reviewer(s) approve.

**Comment by ShivanKaul** [comment]
I'm okay merging into Nightly as long as the privacy policy update is applied. Also, not blocking for Nightly, but [these issues](https://github.com/brave/reviews/issues/1703#issuecomment-2536392931) must be fixed before the feature is enabled beyond.

**Comment by petemill** [comment]
waiting for https://github.com/brave/brave-com/pull/2667

**Pull request merged**